docs/enhancements/0002_day0-integration.md

```diff
@@ -0,0 +1,285 @@
+    `ModuleDay0` when assisted doesn't support such resolution on its own.
+  * if a user wishes to have only some worker nodes modified, he will need to
+    create is own `MachineConfigPool` post cluster installation and then update
+    the `machineconfiguration.openshift.io/role: worker` label of the
+    `MachineConfig` to target the new `MachineConfigPool`.
+    This will make MCO role back the nodes that shouldn't be updated to the
+    payload OS.
+    Using layering for only some of the workers will result in a cluster upgrade
+    that only upgrade the nodes that aren't layered which may cause some
+    inconsistency and confusion
+
+After the `ModuleDay0` was create by KMM, all KMM's features will be used by the
+official KMM's API which is the `ModuleDay0` objects in the cluster.
+
+In case a user will wish to stop the sync between MCO and KMM, he will need to
+remove that label from the `MachineConfig` and KMM will delete its `ModuleDay0`.
+Once done, if the user delete the `MachineConfig` in the cluster then MCO will
+roll-back to the RHCOS image from the payload (without the kmods).
+
+### The `ModuleDay0` CRD
+
+```
+apiVersion: kmm.sigs.x-k8s.io/v1beta1
+kind: ModuleDay0
+spec:
+  osMapping:
+    literal: "Red Hat Enterprise Linux CoreOS 414.92.202312311229-0 (Plow)"
+    containerImage: quay.io/ybettan/rhcos:414.92.202312311229-0
+  machineConfigPoolSelector:
+    node-role.kubernetes.io/worker: ""
+```
+
+* The `containerImage` here must be a full RHCOS image built by correct tools
+  or a container image based on an RHCOS container image and not an image based
+  on UBI.
+
+### Modifying day-0 kmods
+
+In day2, we have `kernelMappings` which defines what image should be use for
+each kernel (or future kernel) in the cluster. Once KMM notice a new kernel on
+some nodes, he will start acting on the relevant `kernelMapping`, for example,
+he will load the new kmod on the node.
+
+In day0 modules, we will take a pro-active approach - we will modify the nodes
+instead of waiting for them to get updated by an external process.
+The reason for that will be clear later on in this doc.
+We will have a single `osMapping` which its output artifcats should exist
+**before** we modify the node.
+
+If the `osMapping` in the `ModuleDay0` is different than the
+`node.status.nodeInfo.osImage` then we override `mc.osImageURL` with the image
+from `moduleday0.spec.osMapping.containerimage`.
+
+MCO will then override the node with the new content and potentialy reboot the
+node.
+
+### Building
+
+If a customer wish us to re-build his kmods, he should add a `build` section
+in the `ModuleDay0` as follow
+```
+apiVersion: kmm.sigs.x-k8s.io/v1beta1
+kind: ModuleDay0
+spec:
+  osMapping:
+    literal: "Red Hat Enterprise Linux CoreOS 414.92.202312311229-0 (Plow)"
+    containerImage: quay.io/ybettan/rhcos:414.92.202312311229-0
+    build:
+      secrets:
+        - name: build-secret
```

Comment:
This will require a namespace as well as `ModuleDay0` doesn't seem to be namespaced.